src/components/Quiz.js

```diff
@@ -1,45 +1,70 @@
 import React, { useState } from "react";
 
+import { Alert } from "react-native";
 import { Button } from "react-native-paper";
 import { TextInput } from "react-native-paper";
 import styled from "styled-components/native";
 
-export const Quiz = ({ selectedDialect }) => {
+export const Quiz = ({ selectedDialect, grammar }) => {
   const [text, setText] = useState("");
+  const [correctAnswer, setCorrectAnswer] = useState("");
+  const [buttonText, setButtonText] = useState("答え合わせ");
+  const [buttonColor, setButtonColor] = useState("#40BA62");
 
-  return (
-    <>
-      <BodyText>Please write this in {selectedDialect.name_en}:</BodyText>
-      <BodyText>お母さんは部屋にいるよ。</BodyText>
-      <TextInput
-        label="Answer here"
-        value={text}
-        onChangeText={(text) => setText(text)}
-        style={{ margin: 20 }}
-      ></TextInput>
-      <ButtonContainer>
-        <Button
-          mode="contained"
-          onPress={() => console.log("button pressed")}
-          color="#40BA62"
-          labelStyle={{ color: "#fff", fontSize: 18 }}
-          style={{
-            width: "50%",
-            height: 45,
-            justifyContent: "center",
-          }}
-        >
-          答え合わせ
-        </Button>
-      </ButtonContainer>
-    </>
-  );
+  if (grammar.quizzes.length > 0) {
+    const quizTokyo = grammar.quizzes[0].tokyo;
+    const answer = grammar.quizzes[0].answer;
+
+    const checkAnswer = (text) => {
+      if (text === answer) {
+        setCorrectAnswer("correct");
```

Comment:
instead of having three separate states here. What if we had a `showCorrectButton` state which could be `true` or `false`?